pkg/console/subresource/deployment/deployment.go

```diff
@@ -367,19 +193,85 @@ func consoleVolumes(vc []volumeConfig) []corev1.Volume {
 			}
 		}
 	}
-	return vols
+	deployment.Spec.Template.Spec.Volumes = vols
 }
 
-func consoleVolumeMounts(vc []volumeConfig) []corev1.VolumeMount {
-	volMountList := make([]corev1.VolumeMount, len(vc))
-	for i, item := range vc {
-		volMountList[i] = corev1.VolumeMount{
-			Name:      item.name,
-			ReadOnly:  item.readOnly,
-			MountPath: item.path,
+func withContainerImage(deployment *appsv1.Deployment, operatorConfig *operatorv1.Console, proxyConfig *configv1.Proxy) {
```

Comment:
Same as above, since it's not deployment agnostic.
```suggestion
func withConsoleContainerImage(deployment *appsv1.Deployment, operatorConfig *operatorv1.Console, proxyConfig *configv1.Proxy) {
```